components/tabs/src/storage.rs

```diff
@@ -105,8 +124,11 @@ impl TabsStorage {
             &crate::schema::TabsMigrationLogic,
         ) {
             Ok(conn) => {
-                self.db_connection = Some(conn);
-                Ok(self.db_connection.as_ref())
+                self.db_connection = DbConnection::Open(conn);
+                match self.db_connection {
+                    DbConnection::Open(ref conn) => Ok(Some(conn)),
+                    _ => Err(Error::UnexpectedConnectionState),
```

Comment:
```suggestion
                    _ => unreachable!("impossible value"),
```

Ditto, this is a great place for an `unreachable!()`.